src/useTimer.ts

```diff
@@ -16,43 +17,38 @@ export interface IValues {
   time: number;
 }
 
-const initialConfig = {
-  endTime: null,
+const initialConfig: IConfig = {
+  endTime: 0,
   initialTime: 0,
   interval: 1000,
   timerType: 'INCREMENTAL',
+  step: 1
 };
 
-export const useTimer = (config?: IConfig): IValues => {
-  const { endTime, initialTime, interval, timerType } = {
+export const useTimer = (config: Partial<IConfig> = initialConfig): IValues => {
+  const { endTime, initialTime, interval, timerType, step } = {
     ...initialConfig,
     ...config
   };
   let intervalRef = useRef<NodeJS.Timeout | null>(null);
-  let [time, setTime] = useState(initialTime);
+  const [time, setTime] = useState(initialTime);
 
   const cancelTimer = () => {
-    if (!intervalRef.current) {
-      return;
+    if (intervalRef.current !== null) {
```

Comment:
With truthiness, no need to explicitly check for null value. And i prefer to short circuit condition by returning first, and putting logic outside if block.